src/components/finder/Finder.tsx

```diff
@@ -23,32 +19,12 @@ const useStyles = makeStyles()(() => ({
 
 export const Finder: FC = observer(() => {
     const { classes } = useStyles()
-    const { data, pathname, loading } = useSharedStore().documentStore
-
-    const [active, setActive] = useState<LocalDocumentData>()
-    const [columns, setColumns] = useState<ColumnItem[]>([])
-
-    useEffect(() => {
-        ;(async () => {
-            if (!loading && pathname && data) {
-                const localData = { ...data }
-                let current: LocalDocumentData | undefined = localData
-                let level = 0
-
-                setActive(current)
-                setColumns(makeColumns(current, getBasePath(pathname)))
-
-                while (current?.parent_id && level <= parentLevels) {
-                    current.parent = await docAPI(getBasePath(pathname) + current.parent_id, current.parent_children_page)
-
-                    current = current.parent
-                    level++
+    const { pathname, hierarchy } = useSharedStore().documentStore
 
-                    setColumns(makeColumns(localData, getBasePath(pathname)))
-                }
-            }
-        })()
-    }, [data, pathname, loading])
+    const columns = useMemo(() => {
```

Comment:
please cache it at the store level

Reply:
I still believe that using `useMemo` within the Finder component is the right approach for a few reasons:

- **Separation of concerns**: We should strive to keep our MobX store focused on state management and avoid including UI-specific logic within it. By adding the computation of the columns into the store, we're adding a piece of UI logic that may not be necessary in other parts of our application. This could potentially limit the reusability of our store.

- **Performance**: `useMemo` is designed for optimizing performance in React components by memoizing expensive computations. The `makeColumns` function is only run when hierarchy or pathname changes, which means we are not doing any unnecessary computations. This is particularly efficient when the computation is only needed in this specific component, as the operation is performed only when the component is rendered.

So while caching this at the store level is definitely possible (as you suggested), I believe the benefits of keeping our store decoupled from UI logic and using React's built-in performance optimizations justify the current implementation.